webapp/packages/plugin-connections/src/ConnectionShield.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observer } from 'mobx-react-lite';
+import { type PropsWithChildren, useState } from 'react';
+
+import { Button, Loader, TextPlaceholder, useResource, useTranslate } from '@cloudbeaver/core-blocks';
+import { ConnectionInfoResource, ConnectionsManagerService, type IConnectionInfoParams } from '@cloudbeaver/core-connections';
+import { useService } from '@cloudbeaver/core-di';
+import { NotificationService } from '@cloudbeaver/core-events';
+
+interface Props {
+  connectionKey: IConnectionInfoParams | null;
+}
+
+export const ConnectionShield = observer<PropsWithChildren<Props>>(function ConnectionShield({ connectionKey, children }) {
+  const translate = useTranslate();
+  const connectionsManagerService = useService(ConnectionsManagerService);
+  const notificationService = useService(NotificationService);
+
+  const connection = useResource(ConnectionShield, ConnectionInfoResource, connectionKey);
+
+  const [connecting, setConnecting] = useState(false);
+
+  async function handleConnect() {
+    if (connecting || !connection.data || !connectionKey) {
+      return;
+    }
+
+    setConnecting(true);
+
+    try {
+      await connectionsManagerService.requireConnection(connectionKey);
+    } catch (exception: any) {
+      notificationService.logException(exception);
+    } finally {
+      setConnecting(false);
+    }
+  }
+
+  if (connection.data && !connection.data.connected) {
```

Comment:
I dont like we have drawing logic here and in the parent component. Lets make it in one place here

we can add extra prop something like `noDataPlaceholderText`

and if there is no `connectcion.data` we draw placeholder here.

if `noDataPlaceholderText` is not provided, draw a default defined placeholder

after that you can safely remove this code from the parent component:
`<TextPlaceholder>{translate('plugin_object_viewer_table_no_items')}</TextPlaceholder>`